src/scripts/plugincatalogpage.js

```diff
@@ -1,78 +1,212 @@
+        var availablePlugins = options.availablePlugins;
+        var installedPlugins = options.installedPlugins;
+        var allPlugins = availablePlugins.filter(function (p__w) {
+            p__w.category = p__w.category || "General";
+            p__w.categoryDisplayName = getHeaderText(p__w.category);
+
+            if (!options.categories || -1 != options.categories.indexOf(p__w.category)) {
+                if (!options.targetSystem || p__w.targetSystem == options.targetSystem) {
+                    return "UserInstalled" == p__w.type;
+                }
+
+                return false;
+            }
+
+            return false;
+        });
+        availablePlugins = allPlugins.sort(function (a__e, b__r) {
+            var aName = a__e.category;
+            var bName = b__r.category;
+
+            if (aName > bName) {
+                return 1;
+            }
+
+            if (bName > aName) {
+                return -1;
+            }
+
+            aName = a__e.name;
+            bName = b__r.name;
+
+            if (aName > bName) {
+                return 1;
+            }
+
+            if (bName > aName) {
+                return -1;
+            }
+
+            return 0;
         });
-        var i, length, plugin, currentCategory, html = "";
+        var i__q;
+        var length;
+        var plugin;
+        var currentCategory;
+        var html = "";
+
         if (!options.categories) {
-            currentCategory = globalize.translate("HeaderTopPlugins"), html += '<div class="verticalSection">', html += '<h2 class="sectionTitle sectionTitle-cards">' + currentCategory + "</h2>";
-            var topPlugins = allPlugins.slice(0).sort(function(a, b) {
-                if (a.installs > b.installs) return -1;
-                if (b.installs > a.installs) return 1;
-                var aName = a.name,
-                    bName = b.name;
-                return aName > bName ? 1 : bName > aName ? -1 : 0
+            currentCategory = globalize.translate("HeaderTopPlugins");
+            html += '<div class="verticalSection">';
+            html += '<h2 class="sectionTitle sectionTitle-cards">' + currentCategory + "</h2>";
+            var topPlugins = allPlugins.slice(0).sort(function (a__t, b__y) {
+                if (a__t.installs > b__y.installs) {
+                    return -1;
+                }
+
+                if (b__y.installs > a__t.installs) {
+                    return 1;
+                }
+
+                var aName = a__t.name;
+                var bName = b__y.name;
+
+                if (aName > bName) {
+                    return 1;
+                }
+
+                if (bName > aName) {
+                    return -1;
+                }
+
+                return 0;
             }).filter(isUserInstalledPlugin);
             html += '<div class="itemsContainer vertical-wrap">';
             var limit = screen.availWidth >= 1920 ? 15 : 12;
```

Comment:
some black magic here...

Reply:
Looks like sorting code to me, first by popular and then by name